File: programovani_v_csharp/cviceni/08-excel/Token.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace task3a;

public abstract record Token
{
}


public record EndOfLineToken : Token
{

}

public record WordToken : Token
{
    public readonly string Word;

    public WordToken(string value)
    {
        Word = value;
    }
}

File: NPRG013_programovani_v_jave/cviceni_hnetynka/justify/Token.java
package com.mypackage.task4;

abstract class Token {
}

class WordToken extends Token {
    private String word;

    public WordToken(String word) {
        this.word = word;
    }

    public String getWord() {
        return word;
    }
}

class EndOfStreamToken extends Token {
}

class EndOfParagraphToken extends Token {
}


